website/views.py

```diff
@@ -241,51 +244,87 @@ def user_payment_types(request):
+    Author: Jordan Nelson & Harper Frankstone
+    Args: product_id - the id of the product to be added to the cart
+    Returns: Redirects user to their shopping cart after a successful add
+    """
+    product_to_add = Product.objects.get(pk=product_id)
 
-# @login_required(login_url='/login')
-# def add_product_to_cart(request, pk):
-#         """
-#     purpose: Allows user to add a product to their cart
-
-#     author: Dara Thomas
-
-#     args:  
-
-#     returns: 
-#     """  
-#     product = models.Product.objects.get(id = pk)
+    try:
+        customer = request.user
+        new_order = Order.objects.get(customer=customer, active=1)
+    except ObjectDoesNotExist:
+        customer = request.user
+        new_order = Order.objects.create(customer=customer, order_date=None, payment_type=None, active=1)
 
+    add_to_ProductOrder = ProductOrder.objects.create(product=product_to_add, order=new_order)
 
+    return HttpResponseRedirect('/cart')
 
+@login_required(login_url='/login')
+def view_cart(request):
+    """
+    Purpose: To view the cart of a customer's products
+    Author: Jordan Nelson & Harper Frankstone
+    Args: None
+    Returns: A list of the products added to a shopping cart and their total
+    """
+    total = 0
 
-# @login_required(login_url='/login')
-# def view_cart(request):
-#     """
-#     purpose: Allows user to view cart and all products they've added to cart
+    try:
+        customer = request.user
+        order_id = Order.objects.get(customer=customer, active=1).id
+    except ObjectDoesNotExist:
+        customer = request.user
+        order_id = Order.objects.create(customer=customer, order_date=None, payment_type=None, active=1)
 
-#     author: Dara Thomas
+    try:
+        products_in_cart = ProductOrder.objects.all().filter(order=order_id)
+    except:
+        return render(request, 'cart.html', { 'total' : total, 'orderid' : order_id } )
 
-#     args:  request -- The full HTTP request object
+    for product in products_in_cart:
+        total += product.product.price
 
-#     returns: rendered view of the cart page, with a list of products that are currently in the user's cart
-#     """        
-#     template_name = 'cart.html' 
-#     products_in_cart =  models.Order.objects.get(user = request.user.id)
-#     return render(request, template_name, {"product": product})
+    return render(request, 'cart.html', { 'products_in_cart' : products_in_cart, 'total' : total, 'orderid' : order_id } )
 
+@login_required(login_url='/login')
+def complete_order_add_payment(request, order_id):
+    """
+    purpose: Allows user to add a payment type to their order and therefore complete and place the order
+    author: Dara Thomas
+    args: order_id - no clue what to do with this
+    returns: a checkout page where the user sees their order total and can select a payment type for their order
+    """
+    if request.method == 'POST':
```

Comment:
Add a check to see if a payment type has been added, if not then redirect the user to the add a payment type form.